Sources/App/Core/Git.swift

```diff
@@ -109,3 +98,50 @@ extension Git {
         let date: Date
     }
 }
+
+
+extension ShellOutCommand {
+
+    static var gitClean: Self {
+        .init(string: "git clean -fdx")
+    }
+
+    static var gitCommitCount: Self {
+        .init(string: "git rev-list --count HEAD")
+    }
+
+    static var gitFetch: Self {
+        .init(string: "git fetch --tags")
+    }
+
+    static var gitFirstCommitDate: Self {
+        .init(string: #"git log --max-parents=0 -n1 --format=format:"%ct""#)
+    }
+
+    static var gitLastCommitDate: Self {
+        .init(string: #"git log -n1 --format=format:"%ct""#)
+    }
+
+    static func gitReset(hard: Bool) -> Self {
+        .init(string: "git reset\(hard ? " --hard" : "")")
+    }
+
+    static func gitReset(to branch: String, hard: Bool) -> Self {
+        .init(string: #"git reset "origin/\#(branch)"\#(hard ? " --hard" : "")"#)
+    }
+
+    static func gitRevisionInfo(reference: Reference, separator: String) -> Self {
+        let safe = Git.sanitizeInput("\(reference)")
+        return .init(string: #"git log -n1 --format=format:"%H\#(separator)%ct" "\#(safe)""#)
+    }
+
+    static func gitShowDate(_ commit: CommitHash) -> Self {
+        let safe = Git.sanitizeInput("\(commit)")
+        return .init(string: #"git show -s --format=%ct "\#(safe)""#)
+    }
+
+    static var gitTag: Self {
+        .init(string: "git tag")
+    }
```

Comment:
There are two reasons for moving all these analysis commands into `ShellOutCommand` extensions:

- making mocking easier
- making the command snapshot more readable

More details below.